Common/Orders/TimeInForce.cs

```diff
@@ -0,0 +1,159 @@
+        }
+
+        /// <summary>
+        /// Returns a <see cref="TimeInForceType"/> value from a <see cref="TimeInForce"/> instance.
+        /// </summary>
+        public static implicit operator TimeInForceType(TimeInForce timeInForce)
+        {
+            return timeInForce.Type;
+        }
+
+        /// <summary>
+        /// Equals operator
+        /// </summary>
+        /// <param name="left">The left operand</param>
+        /// <param name="right">The right operand</param>
+        /// <returns>True if both symbols are equal, otherwise false</returns>
+        public static bool operator ==(TimeInForce left, TimeInForce right)
+        {
+            if (ReferenceEquals(left, right)) return true;
+            if (ReferenceEquals(left, null)) return false;
+            if (ReferenceEquals(right, null)) return false;
+
+            return left.Type == right.Type;
+        }
+
+        /// <summary>
+        /// Not equals operator
+        /// </summary>
+        /// <param name="left">The left operand</param>
+        /// <param name="right">The right operand</param>
+        /// <returns>True if both symbols are not equal, otherwise false</returns>
+        public static bool operator !=(TimeInForce left, TimeInForce right)
+        {
+            return !(left == right);
+        }
+
+        /// <summary>
+        /// Determines whether the specified <see cref="TimeInForce"/> is equal to the current <see cref="TimeInForce"/>.
+        /// </summary>
+        /// <param name="obj">The object to compare with the current object. </param>
+        /// <returns>true if the specified object is equal to the current object; otherwise, false.</returns>
+        public bool Equals(TimeInForce obj)
+        {
+            if (ReferenceEquals(null, obj)) return false;
+            if (ReferenceEquals(this, obj)) return true;
+
+            return Type.Equals(obj.Type);
+        }
+
+        /// <summary>
+        /// Determines whether the specified <see cref="T:System.Object"/> is equal to the current <see cref="T:System.Object"/>.
+        /// </summary>
+        /// <param name="obj">The object to compare with the current object. </param>
+        /// <returns>true if the specified object is equal to the current object; otherwise, false.</returns>
+        public override bool Equals(object obj)
+        {
+            if (ReferenceEquals(null, obj)) return false;
+            if (ReferenceEquals(this, obj)) return true;
+
+            if (obj is TimeInForceType)
+                return (TimeInForceType) obj == Type;
+
+            var other = obj as TimeInForce;
+            return other != null && Type.Equals(other.Type);
+        }
+
+        /// <summary>
+        /// Serves as a hash function for a particular type.
+        /// </summary>
+        /// <returns>A hash code for the current <see cref="TimeInForce"/>.</returns>
+        public override int GetHashCode()
+        {
+            return Type.GetHashCode();
```

Comment:
The more I think about equality implementations the more I think these should be defined as abstract or not defined at all in this class (`Equals` and `GetHashCode`) and the operators `==` and `!=` should pipe through to the `Equals` method to enable proper dynamic dispatch. Thoughts?

Reply:
Agree on the abstract equality methods and dynamic dispatch, if possible I'd try leaving the `==` and `!=` in the base class (to avoid duplication)